packages/admin-graphql-api-utilities/src/index.ts

```diff
@@ -43,26 +50,51 @@ export function parseGidWithParams(gid: string): ParsedGid {
   throw new Error(`Invalid gid: ${gid}`);
 }
 
-export function composeGidFactory(namescape: string) {
-  return function composeGid(
-    key: string,
+export function composeGidFactory<N extends string>(namespace: N) {
+  return function composeGid<T extends string>(
+    key: T,
     id: number | string,
     params: {[key: string]: string} = {},
-  ): string {
-    const gid = `gid://${namescape}/${key}/${id}`;
+  ): Gid<N, T> {
+    const gid = `gid://${namespace}/${key}/${id}`;
     const paramKeys = Object.keys(params);
 
     if (paramKeys.length === 0) {
-      return gid;
+      return gid as Gid<N, T>;
     }
 
     const paramString = new URLSearchParams(params).toString();
-    return `${gid}?${paramString}`;
+    return `${gid}?${paramString}` as Gid<N, T>;
   };
 }
 
 export const composeGid = composeGidFactory('shopify');
 
+export function isGidFactory<N extends string>(namespace: N) {
+  return function isGid<T extends string = unknown>(
+    gid: string,
+    key?: T,
+  ): gid is Gid<N, T> {
+    if (!gid.startsWith(`gid://${namespace}/`)) {
+      return false;
+    }
+
+    try {
+      if (key !== undefined && parseGidType(gid) !== key) {
+        return false;
+      }
+    } catch {
+      return false;
+    }
+
+    // prepends forward slash to help identify invalid id
```

Comment:
I know that we don't need to extract the parameters here, but I wonder if it'd be better to just call `parseGidWithParams` rather than have to reproduce some of its internal logic (and think about keeping them in sync in the future).

Reply:
If you check out the PR description, that's originally what I had haha. I chose to inline the part I needed, though, for two reasons:
1. To decouple the logic (if the parser functions want to do other stuff, they can without affecting this function)
2. To be slightly [more performant](https://stackoverflow.com/a/10940138/11621047) (which is probably not even a real concern here, but it's free so why not I guess haha)

Should I replace it with the parser function?